### PR TITLE
Self-checks

### PR DESCRIPTION
**Pull request by petemounce**
Docs for how we expect to use pre-commit here, and pre-commit for this repo (local only, no CI at this stage).

### CODE REVIEW

**Review by ca-johnson — APPROVED** [review]
LGTM!

**Review by jared-improbable — APPROVED** [review]
LGTM - looking forward to all of this coming together

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  - repo: https://github.com/jumanjihouse/pre-commit-hook-yamlfmt
+    rev: 0.0.10
+    hooks:
+      - id: yamlfmt
+        args:
+          - --mapping=2
+          - --sequence=4
+          - --offset=2
+          - --width=1200    # match .yamllint.yaml; we don't care about line-length.
+        exclude: .+/helm/templates/.+
```
Comment (NickTikhonov) [inline .pre-commit-config.yaml]: I don't think you need this line here

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        args:
+          - --autofix
+      - id: trailing-whitespace
+
+  - repo: https://github.com/syntaqx/git-hooks
+    rev: v0.0.16
+    hooks:
+      - id: forbid-binary
+        # Exclude png and snapshots forbid, we can check these in.
+        exclude: ^.+\.(png|snapshot)$
```
Comment (NickTikhonov) [inline .pre-commit-config.yaml]: I think we can forbid both here (at least .snapshot)

**Review by NickTikhonov — APPROVED** [review]

File: .pre-commit-config.yaml (hunk 2 above)
Comment (petemounce) [reply .pre-commit-config.yaml]: Ta, good spot.

**Pull request merged**